src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -303,5 +311,67 @@ default MethodHandle downcallHandle(MemorySegment symbol, FunctionDescriptor fun
         static Option firstVariadicArg(int index) {
             return new LinkerOptions.FirstVariadicArg(index);
         }
+
+        /**
+         * {@return A linker option used to preserve values that might be overwritten by the runtime
+         *          before they can be read through conventional means}
+         * <p>
+         * A downcall method handle linked with this option will feature an additional {@link MemorySegment}
+         * parameter directly following the target address parameter. This memory segment represents
+         * the off-heap memory into which the preserved values are to be written.
+         *
+         * @param preservedValues the names of the values to preserve.
+         * @see PreserveValue#supported()
+         */
+        static PreserveValue preserveValue(String... preservedValues) {
```

Comment:
I think I prefer `saveValue` - it's simpler, but means more or less the same thing.
Also, I think we should use the plural "saveValues" both in the factory and in the option interface.

Reply:
I also find it a bit odd that I can create a LinkerOption whose preservedValue is unrelated from the supported ones? E.g. I'd like to get an exception early, while it seems like the implementation will only give me an exception when I'm creating the downcall, which seems a tad late.

Reply:
True. I think we can validate the names here. (I think I had that in a previous iteration as well).